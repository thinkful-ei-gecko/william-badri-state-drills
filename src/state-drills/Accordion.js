import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: [{
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
          },
          {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
          },
          {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
          },]
    }
    state = {
        currentlyActive: null, 
        
    }

    handleClick = (index) => {
        this.setState({
            currentlyActive: index
        })
    }

    render() {
        return (
            <ul>
                {this.props.sections.map((section, index) => 
                <li>
                    <button onClick={()=>this.handleClick(index)}>{section.title}</button>
                    {(this.state.currentlyActive === index) && <p>{section.content}</p>}
                </li>)}
            </ul>
        );
    }
}


export default Accordion;