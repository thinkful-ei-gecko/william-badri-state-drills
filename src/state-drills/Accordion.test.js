import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe(`Accordion component`, () =>  { 
    const sectionsProp = [{
        title: 'Section 1',
        content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
      },
      {
        title: 'Section 2',
        content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
      },
      {
        title: 'Section 3',
        content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
      }];

    it('renders an empty li when sections is not supplied', () => {
            const wrapper = shallow(<Accordion sections={[]}/>)
            expect(toJson(wrapper)).toMatchSnapshot()
  })

    it('component renders no sections as active by default', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
  })

    it('component opens a clicked section', (index) => {
        const wrapper = shallow(<Accordion section={sectionsProp}/>)
        wrapper.find('button').at(index).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
  })

    it('component only opens the last section when multiple sections have been clicked', () => {
        const wrapper = shallow(<Accordion />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
  })
})