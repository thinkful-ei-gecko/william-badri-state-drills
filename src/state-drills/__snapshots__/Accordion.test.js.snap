// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component component opens a clicked section 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders an empty li when sections is not supplied 1`] = `<ul />`;
